### PR TITLE
Migrate `BloodSugarEntrySheet` to use `BaseBottomSheet`

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.shortcut.com/simpledotorg/story/8920/migrate-bloodsugarentrysheet-to-use-basebottomsheet

### CODE REVIEW

**Comment by msasikanth** [comment]
Make sure to test this out on the device. I have tested, adding, updating, removing blood sugar and also changing the measurement type.

File: app/src/main/java/org/simple/clinic/main/TheActivityComponent.kt [hunk 1]
```diff
@@ -149,7 +152,10 @@ interface TheActivityComponent :
     NextAppointmentCardView.Injector,
     CriticalAppUpdateDialog.Injector,
     EnterDrugStockScreen.Injector,
-    OverdueSearchScreen.Injector {
+    OverdueSearchScreen.Injector,
+    BloodSugarEntrySheet.Injector,
+    ConfirmRemoveBloodSugarDialogInjector,
+    BloodSugarUnitSelectionDialogInjector {
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/main/TheActivityComponent.kt:158]: Why have we added `BloodSugarUnitSelectionDialogInjector` in this PR?

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks good. Left one comment and PTAL at android tests.

File: app/src/main/java/org/simple/clinic/main/TheActivityComponent.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/main/TheActivityComponent.kt:158]: Previously it was part of the `BloodSugarEntryComponent`. So, we have to move that as well

**Comment by Honey14** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4157** <a href="https://app.graphite.dev/github/pr/simpledotorg/simple-android/4157" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/simpledotorg/simple-android/4157?utm_source=stack-comment).

**Review by Honey14 — APPROVED** [review]

**Pull request merged**